iohk/iohk-ops.hs

```diff
@@ -0,0 +1,276 @@
+                 <$> (optional centralCommandParser) <*> (optional centralCommandParser) <*> (optional centralCommandParser) <*> (optional centralCommandParser))
+
+
+-- * Central command
+--
+data Command where
+
+  -- * setup 
+  Template              :: { tNodeLimit   :: Integer
+                           , tEnvironment :: Environment
+                           , tTarget      :: Target
+                           , tBranch      :: Branch
+                           , tDeployments :: [Deployment]
+                           } -> Command
+  SetCardano            :: Commit -> Command
+  SetExplorer           :: Commit -> Command
+  MiniKeys              :: Command
+
+  -- * building
+  Genesis               :: Command
+  GenerateIPDHTMappings :: Command
+  Build                 :: Deployment -> Command
+  AMI                   :: Command
+
+  -- * cluster lifecycle
+  Nixops                :: NixopsCmd -> [Text] -> Command
+  Do                    :: [Command] -> Command
+  Create                :: Command
+  Modify                :: Command
+  Deploy                :: Command
+  Destroy               :: Command
+  Delete                :: Command
+  FromScratch           :: Command
+  Info                  :: Command
+
+  -- * live cluster ops
+  CheckStatus           :: Command
+  Start                 :: Command
+  Stop                  :: Command
+  FirewallBlock         :: { from :: Region, to :: Region } -> Command
+  FirewallClear         :: Command
+  RunExperiment         :: Deployment -> Command
+  PostExperiment        :: Command
+  DumpLogs              :: { depl :: Deployment, withProf :: Bool } -> Command
+  PrintDate             :: Command
+deriving instance Show Command
+
+centralCommandParser :: Parser Command
+centralCommandParser =
+  (    subcommandGroup "General:"
+    [ ("template",              "Clone iohk-nixops from git BRANCH, for a specified set of deployments",
+                                Template
+                                <$> (fromMaybe Ops.defaultNodeLimit
+                                     <$> optional (optInteger "node-limit" 'l' "Limit cardano-node count to N"))
+                                <*> parserEnvironment
+                                <*> parserTarget
+                                <*> parserBranch "iohk-nixops branch to check out"
+                                <*> parserDeployments)
+    , ("set-cardano",           "Set cardano-sl commit to COMMIT",
+                                SetCardano  <$> parserCommit "Commit to set 'cardano-sl' version to")
+    , ("set-explorer",          "Set cardano-sl-explorer commit to COMMIT",
+                                SetExplorer <$> parserCommit "Commit to set 'cardano-sl-explorer' version to")
+    , ("mini-keys",             "Fake/enter minimum set of keys necessary for a basic deployment",  pure MiniKeys)
+    , ("do",                    "Chain commands",                                                   Do <$> parserDo) ]
```

Comment:
The docs (https://github.com/input-output-hk/internal-documentation/wiki/iohk-ops-reference#general) provide one:
Instead of:
```
io -c staging.yaml destroy && io -c staging.yaml delete && io -c staging.yaml create && io -c staging.yaml deploy
```
..we have:
```
io -c staging.yaml do destroy delete create deploy
```

Of course we have `fromscratch`, but this command is generic.